crates/op-rbuilder/src/builders/flashblocks/payload.rs

```diff
@@ -207,11 +207,28 @@ where
     }
 }
 
+/// The arena in which multiple [`FlashblockCandidate`] are competing for a specific flashblock interval batch.
+/// It contains context of the building job, and shared precomputed elements that are required in each candidate.
+struct BatchArena {
```

Comment:
I would rather keep it more lean, have best_flashblock variable, implement Ord for FlashblockPayload

Reply:
Or custom function for comparison

Reply:
done